prometheus/promhttp/http.go

```diff
@@ -381,6 +393,35 @@ type HandlerOpts struct {
 	ProcessStartTime time.Time
 }
 
+type FilteredDefaultGatherer struct {
```

Comment:
I would just add `FilteredGatherer` that wraps registry and pass through only matching metrics. Matchers can be done with something like https://github.com/prometheus/client_golang/blob/main/prometheus/collectors/go_collector_latest.go#L93 - regexp. 

This allows us to allow list those metrics you wanted to hardcode here, outside of `FilteredGatherer` - on caller side (: